### ISSUE TITLE
1.0.0-pre iOS BannerAds keep floating on top or at bottom when added into ListView while scrolling

### ISSUE THREAD

**Issue by ad-on-is** [bdlukaa/native_admob_flutter#22]
I just encountered that on iOS the BannerAd, embedded in a ListView, would stay on top, or at the bottom when scrolled away from the view.

```dart
    ListView(
      children: [SomeWidget(), BannerAd(), SomeWidget(), SomeWidget(), BannerAd(), SomeWidget()]
    )
```

![image](https://user-images.githubusercontent.com/9553253/110211952-58cf2a80-7e99-11eb-8144-61e9d70aabbc.png)

**Comment by bdlukaa** [comment]
This is actually an issue with hybrid composition. The same happens when using hybrid composition on Android.  (hybrid composition is the only option on iOS)

Can you try running it on release mode (`flutter run --release`) and see if it works?

**Comment by ad-on-is** [comment]
Same issue in release mode

**Comment by bdlukaa** [comment]
I couldn't reproduce this issue. Everything works fine.

**Comment by sharokh1** [comment]
I have the same.. something is not going well with listviews and ads.
check --> https://prnt.sc/10o7fjd

**Comment by bdlukaa** [comment]
This is an issue with hybrid composition. The same happens when using hybrid composition on Android. Unfortunately hybrid composition is the only option on iOS.

**Comment by bdlukaa** [comment]
One solution is to use a `Texture`. I'll work on this

**Comment by bdlukaa** [comment]
Yeah a `Texture` is not a viable thing, because the ads needs to be clickable, but a Texture just paint the view without any gesture detector. Unfotunately this doesn't depends on me, but in the flutter team. Probably the same will happen with other plugins that use platform views, such as [google_mobile_ads](https://pub.dev/packages/google_mobile_ads) and [webview_flutter](https://pub.dev/packages/webview_flutter).

**Comment by sharokh1** [comment]
Hmm I have another project in flutter using the flutter_google_ad_manager for the banners and I do not have the same issue there... i'll check what the differences are.

**Comment by bdlukaa** [comment]
I do not recommend using ads in a ListView in iOS until a viable option arrives. As the purpose of banner ads is to be on the top or in the bottom of the screen, you can add it there. See the example project for reference

**Comment by sharokh1** [comment]
The fix for me is not to put the ad in a Column inside a ListView!

**Comment by eliabieri** [comment]
My solution was to place the Ad widget at the top of the listview.

**Comment by Tetracyl** [comment]
I am also experiencing the same issue. There are a couple of issues on the Flutter repo regarding this problem. Hopefully, it can get fixed soon.

https://github.com/flutter/flutter/issues/78205
https://github.com/flutter/flutter/issues/76097

**Comment by bdlukaa** [comment]
@Tetracyl None of the solutions above worked for you?

> My solution was to place the Ad widget at the top of the listview.

> The fix for me is not to put the ad in a Column inside a ListView!

**Comment by Tetracyl** [comment]
The same issue occurs with ClipRect. 

I want to have multiple ads appear throughout one page. So if I place my ad at the top of a listview or take it out of a column then I cannot have multiple ads scattered throughout the page.

**Comment by bdlukaa** [comment]
Can you try using the master channel to see if it fixes the issue? https://github.com/flutter/flutter/issues/76097#issuecomment-817635743

**Comment by Tetracyl** [comment]
I can't since I already updated my app to Flutter 2.0

**Comment by bdlukaa** [comment]
I think this will be closed whenever https://github.com/flutter/engine/pull/22336 lands on stable. For now, I think you guys can use the master channel as a fix for it.

**Comment by bdlukaa** [comment]
This is currently blocked by https://github.com/flutter/flutter/issues/76097

**Comment by bdlukaa** [comment]
For now, you guys can try using Flutter 1.2.6. You can still use null-safety with it.

https://github.com/flutter/flutter/issues/76097#issuecomment-810009590

**Comment by bdlukaa** [comment]
This issue was fixed on flutter version 2.2. See https://github.com/flutter/flutter/issues/76097 for more
